src/status_im/contexts/wallet/wallet_connect/modals/sign_message/view.cljs

```diff
@@ -12,6 +14,43 @@
             [utils.i18n :as i18n]
             [utils.re-frame :as rf]))
 
+(def ^:constant typed-data-methods
+  #{constants/wallet-connect-eth-sign-typed-v4-method
+    constants/wallet-connect-eth-sign-typed-method})
+
+(defn typed-data-field
+  [{:keys [label value]}]
+  [quo/data-item
+   {:card?           false
+    :container-style (merge style/data-item
+                            {:margin-bottom 4})
+    :title           label
+    :subtitle        value}])
+
+(defn typed-data-view
+  []
+  (let [display-data (rf/sub [:wallet-connect/current-request-display-data])]
+    [gesture/flat-list
+     {:data                            display-data
+      :style                           style/data-border-container
+      :over-scroll-mode                :never
+      :content-container-style         {:padding-bottom 12}
+      :render-fn                       typed-data-field
+      :shows-vertical-scroll-indicator false}]))
+
+(defn message-data-view
+  []
+  (let [display-data (rf/sub [:wallet-connect/current-request-display-data])]
+    [data-block/view display-data]))
+
+(defn display-data-view
+  []
+  (let [typed-data? (->> (rf/sub [:wallet-connect/current-request-method])
+                         (contains? typed-data-methods))]
```

Comment:
Could be a separate subscription. `typed-data-methods` is static, so the result for `typed-data?` will be nicely cached.